File: main/symmetric-tree/symmetric-tree-bfs.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    @staticmethod
    def isSymmetric(root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root is None:
            return True

        queue = collections.deque()
        queue.append((root.left, root.right))

        while queue:
            left, right = queue.popleft() # this popleft is unrelated to "left"
            if left is None or right is None:
                if left is not None or right is not None:
                    return False
            elif left.val != right.val:
                return False
            else:
                queue.append((left.left, right.right))
                queue.append((left.right, right.left))

        return True

File: main/symmetric-tree/symmetric-tree-bfs.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsSymmetric(TreeNode root) {
        if (root == null) return true;

        var queue = new Queue<(TreeNode, TreeNode)>();
        queue.Enqueue((root.left, root.right));

        while (queue.Count != 0) {
            var (left, right) = queue.Dequeue();

            if (left == null || right == null) {
                if (left != null || right != null) return false;
            }
            else if (left.val != right.val) return false;
            else {
                queue.Enqueue((left.left, right.right));
                queue.Enqueue((left.right, right.left));
            }
        }

        return true;
    }
}
